mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/KeyValueBasedPersistenceProvider.java

```diff
@@ -561,7 +569,14 @@ public Optional<IMantisJobMetadata> loadArchivedJob(String jobId) throws IOExcep
                 for (IMantisWorkerMetadata w : archivedWorkers) {
                     try {
                         MantisWorkerMetadataWritable wmw = DataFormatAdapter.convertMantisWorkerMetadataToMantisWorkerMetadataWritable(w);
-                        jmw.addWorkerMedata(w.getStageNum(), wmw, null);
+                        Preconditions.checkState(
```

Comment:
I'd avoid throwing an exception for archived job (or even archived worker) since the job isn't already running.

Reply:
I got rid of this code fully. WIll it have any impact?